src/librustdoc/passes/unindent_comments.rs

```diff
@@ -35,65 +34,72 @@ impl clean::Attributes {
+                // level of the first line
+                let ignore_previous_indents =
+                    saw_first_line && !saw_second_line && !line.chars().all(|c| c.is_whitespace());
 
-        if saw_first_line {
-            saw_second_line = true;
-        }
+                let min_indent = if ignore_previous_indents { usize::MAX } else { min_indent };
+
+                if saw_first_line {
+                    saw_second_line = true;
+                }
 
-        if line.chars().all(|c| c.is_whitespace()) {
-            min_indent
-        } else {
-            saw_first_line = true;
-            let mut whitespace = 0;
-            line.chars().all(|char| {
-                // Compare against either space or tab, ignoring whether they
-                // are mixed or not
-                if char == ' ' || char == '\t' {
-                    whitespace += 1;
-                    true
+                if line.chars().all(|c| c.is_whitespace()) {
+                    min_indent
                 } else {
-                    false
+                    saw_first_line = true;
+                    let mut whitespace = 0;
+                    line.chars().all(|char| {
+                        // Compare against either space or tab, ignoring whether they
+                        // are mixed or not
+                        if char == ' ' || char == '\t' {
+                            whitespace += 1;
+                            true
+                        } else {
+                            false
+                        }
+                    });
+                    cmp::min(min_indent, whitespace)
                 }
-            });
-            cmp::min(min_indent, whitespace)
-        }
-    });
+            })
+        })
+        .collect::<Vec<_>>();
+    if min_indents.is_empty() {
+        return;
+    }
+    min_indents.sort_unstable();
+    let min_indent = min_indents[0];
 
-    if !lines.is_empty() {
-        let mut unindented = vec![lines[0].trim_start().to_string()];
-        unindented.extend_from_slice(
-            &lines[1..]
-                .iter()
-                .map(|&line| {
-                    if line.chars().all(|c| c.is_whitespace()) {
-                        line.to_string()
-                    } else {
-                        assert!(line.len() >= min_indent);
-                        line[min_indent..].to_string()
-                    }
-                })
-                .collect::<Vec<_>>(),
-        );
-        unindented.join("\n")
-    } else {
-        s.to_string()
+    for (index, fragment) in docs.iter_mut().enumerate() {
+        let lines = fragment.doc.lines().collect::<Vec<&str>>();
+
+        if !lines.is_empty() {
+            let mut unindented =
+                if index == 0 { vec![lines[0].trim_start().to_string()] } else { vec![] };
+            unindented.extend_from_slice(
+                &lines[if index == 0 { 1 } else { 0 }..]
```

Comment:
I don't understand what this is doing, can you add a comment?

Reply:
Actually, I think that most of this code can be simplified now.